### PR TITLE
[Phone] update to latest SDK

### PR DESCRIPTION
**Pull request by bobokids**
### Purpose
*What is the context of this pull request? Why is it being done?*
This change is to update phone skill to 4.8.0-rc4.

### Changes
*Are there any changes that need to be called out as significant or particularly difficult to grasp? (Include illustrative screenshots for context if applicable.)*
. SDK alignment
. LG update to localTemplateManager and new lg structure

### Tests
*Is this covered by existing tests or new ones? If no, why not?*

### Feature Plan
*Are there any remaining steps or dependencies before this issue can be fully resolved? If so, describe and link to any relevant pull requests or issues.*

#### General
- [ ] I have commented my code, particularly in hard-to-understand areas	
- [ ] I have added or updated the appropriate tests	
- [ ] I have updated related documentation

### CODE REVIEW

File: skills/csharp/tests/phoneskill.tests/Flow/PhoneSkillTestBase.cs [hunk 1]
```diff
@@ -52,10 +51,12 @@ public class PhoneSkillTestBase : BotTestBase
 
         public IServiceManager ServiceManager { get; set; }
 
-        public LocaleTemplateEngineManager TemplateEngine { get; set; }
+        public LocaleTemplateManager TemplateEngine { get; set; }
```
Comment (lzc850612) [inline skills/csharp/tests/phoneskill.tests/Flow/PhoneSkillTestBase.cs]: TemplateEngine -> TemplateManager

File: skills/csharp/tests/phoneskill.tests/PhoneSkill.Tests.csproj [hunk 2]
```diff
@@ -1,10 +1,10 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>netcoreapp3.0</TargetFramework>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
     <IsPackable>false</IsPackable>
     <NoWarn>NU1701</NoWarn>
-    <Platforms>AnyCPU;x64</Platforms>
+    <Platforms>AnyCPU;x64;x86</Platforms>
```
Comment (xieofxie) [inline skills/csharp/tests/phoneskill.tests/PhoneSkill.Tests.csproj]: not x86?

File: skills/csharp/shared/SkillServiceLibrary/SkillServiceLibrary.csproj [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>netcoreapp3.0</TargetFramework>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
```
Comment (xieofxie) [inline skills/csharp/shared/SkillServiceLibrary/SkillServiceLibrary.csproj]: not this. Will break pr validation

File: skills/csharp/tests/phoneskill.tests/PhoneSkill.Tests.csproj (hunk 2 above)
Comment (bobokids) [reply skills/csharp/tests/phoneskill.tests/PhoneSkill.Tests.csproj]: fixed.

File: skills/csharp/tests/phoneskill.tests/Flow/PhoneSkillTestBase.cs (hunk 1 above)
Comment (bobokids) [reply skills/csharp/tests/phoneskill.tests/Flow/PhoneSkillTestBase.cs]: fixed

File: skills/csharp/shared/SkillServiceLibrary/SkillServiceLibrary.csproj (hunk 3 above)
Comment (bobokids) [reply skills/csharp/shared/SkillServiceLibrary/SkillServiceLibrary.csproj]: fixed

**Comment by xieofxie** [comment]
Forget update BotServices.cs

File: skills/csharp/experimental/phoneskill/Responses/Shared/PhoneSharedResponses.lg [hunk 4]
```diff
@@ -1,54 +1,57 @@
...
 ]
 
 # AuthFailed(Data, Cards, Layout)
 [Activity
-    Text = @{AuthFailed.Text(Data)}
-    Speak = @{AuthFailed.Text(Data)}
-    Attachments = @{if(Cards == null, null, foreach(Cards, Card, CreateCard(Card)))}
-    AttachmentLayout = @{if(Layout == null, 'list', Layout)}
+    Text = ${AuthFailed.Text(Data)}
+    Speak = ${AuthFailed.Text(Data)}
```
Comment (KayMKM) [inline skills/csharp/experimental/phoneskill/Responses/Shared/PhoneSharedResponses.lg]: AuthFailed have multiple texts

Comment (lzc850612) [reply skills/csharp/experimental/phoneskill/Responses/Shared/PhoneSharedResponses.lg]: LG should be able to just pick one from the list and evaluate right?

**Pull request merged**